### PR TITLE
Release notes for operator-sdk version upgrade epic

### PR DESCRIPTION
**Pull request by Jaanki**
https://github.com/submariner-io/enhancements/issues/81

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

**Review by skitt — CHANGES_REQUESTED** [review]
I thought we’d said the Operator SDK changes didn’t need release notes, since they’re not user-visible. Has that changed?

**Comment by Jaanki** [comment]
I wasn't aware of this decision. Anyways, I think the path change for APIs should be mentioned. It could be used by other projects.

File: release-notes/20221103-operator-sdk-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<!-- markdownlint-disable MD041 -->
+* operator-sdk version is upgraded to 1.23.0.
+* Submariner-operator APIs are converted to single-group. Meaning, API definitions can now be found under `api/`.
```
Comment (skitt) [inline release-notes/20221103-operator-sdk-upgrade.md]:
```suggestion
* The project was rebuilt using Operator SDK version 1.23.0.
* The operator APIs have been converted to single-group; projects depending on these will need to
  replace `github.com/submariner-io/submariner-operator/api/submariner/v1alpha1` with
  `github.com/submariner-io/submariner-operator/api/v1alpha1` in their imports.
```

**Review by skitt — CHANGES_REQUESTED** [review]

**Comment by skitt** [comment]
> I wasn't aware of this decision. Anyways, I think the path change for APIs should be mentioned. It could be used by other projects.

Fair enough, let’s describe the impact explicitly.

**Comment by tpantelis** [comment]
> > I wasn't aware of this decision. Anyways, I think the path change for APIs should be mentioned. It could be used by other projects.
> 
> Fair enough, let’s describe the impact explicitly.

Also discussing this on slack. We've stated and agreed in the past that RNs are for end users. A Go API change is not for end users. We've many Go API changes in the past sans RN. Let's stay consistent.

**Comment by tpantelis** [comment]
> Also discussing this on slack. We've stated and agreed in the past that RNs are for end users. A Go API change is not for end users. We've many Go API changes in the past sans RN. Let's stay consistent.

As a follow-up, to notify possible users of the **submariner-operator** project of the API change, we could add a note to the project release tag once that's created.

**Pull request closed without merging**